File: python/atelier_bench/rules/revenue_floor.py
"""revenue-floor rule (nominal: subclasses RuleContract)."""

from __future__ import annotations

from typing import TYPE_CHECKING

from ..contracts.rule_contract import RuleContract

if TYPE_CHECKING:
    from ..dataset import Dataset


class RevenueFloorRule(RuleContract):
    KEY = "revenue-floor"

    def __init__(self) -> None:
        self.key = self.KEY

    def evaluate(self, data: "Dataset") -> bool:
        return data.revenue_cents() >= 58325

File: go/atelier/rules/revenue_floor.go
// Package rules carries the 48 rules: 24 NOMINAL rule structs in this package
// (each with a compile-time assertion against RuleContract and a value
// receiver) plus the 24 STRUCTURAL ad-hoc rule values from rules/adhoc, which
// assert nothing and satisfy the same method set by shape alone.
package rules

import (
	"atelier.example/lane/atelier/contracts"
	"atelier.example/lane/atelier/dataset"
)

// Compile-time proof against the NOMINAL contract, asserted on the VALUE
// type: RevenueFloorRule's methods take value receivers, so both RevenueFloorRule and
// *RevenueFloorRule satisfy RuleContract.
var _ contracts.RuleContract = RevenueFloorRule{}

// RevenueFloorRule is the revenue-floor rule.
type RevenueFloorRule struct{}

// Key is the rule identifier.
func (RevenueFloorRule) Key() string {
	return "revenue-floor"
}

// Evaluate reports whether the rule holds on this dataset.
func (r RevenueFloorRule) Evaluate(data *dataset.Dataset) bool {
	return data.RevenueCents() >= 58325
}

// init registers this rule.
func init() {
	Register(RevenueFloorRule{})
}
